cmd/controller-manager/app/helper_test.go

```diff
@@ -60,6 +60,20 @@ func TestIsControllerEnabled(t *testing.T) {
 			disabledByDefaultControllers: []string{"delta", "echo"},
 			expected:                     false,
 		},
+		{
+			name:                         "on by all except",
+			controllerName:               "alpha",
+			controllers:                  []string{"*", "-charlie"},
+			disabledByDefaultControllers: []string{"delta", "echo"},
+			expected:                     true,
+		},
```

Comment:
does it diff with the case[on by default]?

Reply:
`on by all except` tests `all but not disabled by sign(-)` in `controllers`.

`on by default` tests all but not in `disabledByDefaultControllers`

They are different by where disable, one in `controllers`, one in `disabledByDefaultControllers`